test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -1080,6 +1080,7 @@ class XdsServerSecurityTest : public XdsEnd2endTest {
 
   void SendRpc(
```

Comment:
I don't think we need a channel creator callback in the first place.  That might have made sense when we were creating a new channel for each RPC, but now that every RPC is using the same channel, it's really exactly the same functionality as `SendRpcsUntil()`.  In fact, I notice that there are a lot of tests where there are two successive calls to `SendRpc()` that have exactly the same channel creator callback, which means that we're needlessly recreating the channel.

(I would also note that for the cases where there are two successive calls to `SendRpc()` with *different* channel creator callbacks, those cases are not necessarily realistic, because there's no way you could deploy such a change safely on the server side, since the clients would all have to be updated atomically at the same time.  But I haven't looked at every test case exhaustively to see if there's still value in covering those scenarios for other reasons.)

I think we should instead just call `ResetStub()` with the appropriate channel args and credentials.

Reply:
> (I would also note that for the cases where there are two successive calls to SendRpc() with different channel creator callbacks, those cases are not necessarily realistic, because there's no way you could deploy such a change safely on the server side, since the clients would all have to be updated atomically at the same time. But I haven't looked at every test case exhaustively to see if there's still value in covering those scenarios for other reasons.)

I do cover those kinds of tests.. mtls to fallback, mtls to tls, tls to mtls and such. I don't know if people will actually make such transactions but I have tests for these anyway.